src/callbacks/amr.jl

```diff
@@ -0,0 +1,302 @@
+
+"""
+    AMRCallback(semi, indicator [,adaptor=AdaptorAMR(semi)];
+                interval=5,
+                adapt_initial_conditions=true,
+                adapt_initial_conditions_only_refine=true)
+
+Performs adaptive mesh refinement (AMR) every `interval` time steps
+for a given semidiscretization `semi` using the chosen `indicator`.
+"""
+struct AMRCallback{Indicator, Adaptor}
+  indicator::Indicator
+  interval::Int
+  adapt_initial_conditions::Bool
+  adapt_initial_conditions_only_refine::Bool
+  adaptor::Adaptor
+end
+
+
+function AMRCallback(semi, indicator, adaptor; interval=5,
```

Comment:
The current default is zero 
https://github.com/trixi-framework/Trixi.jl/blob/2c670e2b0f14e431428923ede8df90e130f9c158/src/run.jl#L121
I changed the Taal default accordingly in c77f2d6e13eba011321d98436459e42452784276. 

I just used `5` since we used that in many examples before :sweat_smile: The rational behind choosing a default and making it a keyword argument is that the user interface is more clear, I think. It's not necessarily clear what `5` means in `AMRCallback(5, ...)`, but it's more clear in `AMRCallback(..., interval=5)`.

Reply:
I think a kwarg is a good choice! However, why provide a default value at all? If someone inexperience creates the AMR callback and does not provide `interval`, it now will do... nothing. I think it would be better to throw an error and force the user to pick something, wouldn't it?

(Before, a default of 0 made sense since it was our way of saying "don't use AMR", but now you can achieve this by just not creating an AMR callback...)

Reply:
Done in dfcdde0